core/blockchain.go

```diff
@@ -445,7 +445,11 @@ func (bc *BlockChain) repair(head **types.Block) error {
 			return nil
 		}
 		// Otherwise rewind one block and recheck state availability there
-		(*head) = bc.GetBlock((*head).ParentHash(), (*head).NumberU64()-1)
+		block := bc.GetBlock((*head).ParentHash(), (*head).NumberU64()-1)
+		if block == nil {
+			return fmt.Errorf("failed to repair block, can not get block at height %d", (*head).NumberU64())
```

Comment:
Lets make this error a bit shorter:

`return fmt.Errorf("missing block %d [%x]", (*head).NumberU64()-1, (*head).ParentHash())`

Reply:
ok, I have fix it. thanks :)

Reply:
@karalabe fixed, thanks